presto-main/src/main/java/com/facebook/presto/operator/aggregation/EntropyAggregation.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.EntropyState;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.AggregationState;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+
+/**
+ *  Calculates the log-2 entropy of count inputs.
+ *
+ *  Given counts $c_1, c_2, ..., cn$ ($c_i \geq 0)$, this aggregation calculates $\sum_i [ p_i log_2(-p_i) ]$,
+ *      where $p_i = {c_i \over \sum_i [ c_i ]}$. If $\sum_i [ c_i ] = 0$, the entropy is defined to be 0.
+ */
+@AggregationFunction("entropy")
+@Description("Computes the log-2 entropy of counts inputs")
+public final class EntropyAggregation
+{
+    /**
+     * @note If count is negative, the value of the aggregation will be null; if
+     * count is 0, this is a no-op (since, in the context of entropy, 0 log(0) = 0; if count is null,
+     * this is a no op.
+     */
+    @InputFunction
+    public static void input(
+            @AggregationState EntropyState state,
+            @SqlType(StandardTypes.BIGINT) long count)
+    {
+        if (count < 0) {
+            state.setNull(true);
+            return;
+        }
+
+        if (count == 0) {
+            return;
+        }
+        state.setSumC(state.getSumC() + count);
+        state.setSumCLogC(state.getSumCLogC() + count * Math.log(count));
+    }
+
+    @CombineFunction
+    public static void combine(@AggregationState EntropyState state, @AggregationState EntropyState otherState)
+    {
+        state.setNull(state.getNull() || otherState.getNull());
+        if (state.getNull()) {
+            return;
+        }
+
+        state.setSumC(state.getSumC() + otherState.getSumC());
+        state.setSumCLogC(state.getSumCLogC() + otherState.getSumCLogC());
+    }
+
+    @OutputFunction(StandardTypes.DOUBLE)
+    public static void output(@AggregationState EntropyState state, BlockBuilder out)
+    {
+        final Double entropy = EntropyAggregation.calculateEntropy(state);
```

Comment:
Not sure why you want to have a separate method for this. But if you think it's better to have a separate method, import static and use `Double entropy = calculateEntropy(state)`. Presto convention is to use static import unless it's too confusing. If no other class will call `calculateEntropy`, it should be a private method.